### PR TITLE
New feature for evaluation

### PR DESCRIPTION
**Pull request by Reindulger**
## Description

- Added evaluation support for text-to-video and text-to-audio modalities.
- Enhanced the text-to-image modality with FID and IS metrics, evaluated on the MSCOCO-2014 benchmark.
- Optimized the evaluation process for existing benchmarks.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [[CONTRIBUTION](https://github.com/PKU-Alignment/align-anything/blob/HEAD/.github/CONTRIBUTING.md)](https://github.com/PKU-Alignment/align-anything/blob/HEAD/.github/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by XuyaoWang — CHANGES_REQUESTED** [review]

File: align_anything/evaluation/README.md [hunk 1]
```diff
@@ -10,9 +10,9 @@ Then, We have a roadmap for future development work `align-anything`:
...
-| **Text ➡️ Image**      | [ImageReward](https://huggingface.co/datasets/THUDM/ImageRewardDB), [HPSv2](https://huggingface.co/datasets/zhwang/HPDv2)➖                                                            |
-| **Text ➡️ Video**      | ➖                                                            |
-| **Text ➡️ Audio**      | ➖                                                            |
+| **Text ➡️ Image**      | [HPSv2](https://huggingface.co/datasets/zhwang/HPDv2), [MSCOCO (ImageReward, FID, IS)](https://huggingface.co/datasets/nlphuji/mscoco_2014_5k_test_image_text_retrieval), ➖ |
```
Comment (XuyaoWang) [inline align_anything/evaluation/README.md:13]: Why use MSCOCO (ImageReward, FID, IS)? ImageReward has its evaluation prompts

File: scripts/eval_with_local_data.sh
Comment (XuyaoWang) [inline scripts/eval_with_local_data.sh:1]: evaluate_with_local_data.sh

**Comment by htlou** [comment]
LGTM

**Review by htlou — COMMENTED** [review]
LGTM. There's some conflict though. You should fix them before merging this pr.

File: scripts/eval_with_local_data.sh
Comment (Reindulger) [reply scripts/eval_with_local_data.sh:1]: get it

**Comment by Reindulger** [comment]
> LGTM. There's some conflict though. You should fix them before merging this pr.

OK. I'll fix them.

File: align_anything/evaluation/README.md (hunk 1 above)
Comment (Reindulger) [reply align_anything/evaluation/README.md:13]: This is the old version of README. I'll change it to the new version.

File: align_anything/configs/evaluation/benchmarks/audiocaps.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    # Num shot
+    n_shot: 0
+    # Use Chain of Thought
+    cot: false
+  # Configuration for data
+  data_cfgs:
+    # Task name
+    task: default
+    # Task directory
+    task_dir: null
```
[the remark this answers is not in the document]
Comment (Reindulger) [reply align_anything/configs/evaluation/benchmarks/audiocaps.yaml:39]: Yes, I'll add it.

File: align_anything/configs/evaluation/benchmarks/mscoco.yaml [hunk 3]
```diff
@@ -27,14 +27,17 @@ default:
...
+    # Use Chain of Thought
+    cot: false
   # Configuration for data
   data_cfgs:
     # Task name
-    task: all
+    task: TEST
     # Task directory
-    task_dir: PKU-Alignment/eval-anything500
+    task_dir: nlphuji/mscoco_2014_5k_test_image_text_retrieval
```
[the remark this answers is not in the document]
Comment (Reindulger) [reply align_anything/configs/evaluation/benchmarks/mscoco.yaml]: I've replaced it.

**Review by htlou — APPROVED** [review]
LGTM, I also fixed some of the existing problems. Looks like this can be merged now :)

**Review by XuyaoWang — APPROVED** [review]
LGTM.

**Pull request merged**